### PR TITLE
feat(gas-rebate): Add script to calculate gas rebates for DVM voting

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

Smaller UMA holders are disincentivized to participate in votes if rewards to not outweigh gas costs, currently this prices out many UMA holders and particularly the new ones who earned UMA via liquidity mining. This script should be used to calculate gas rebates.

**Summary**

Rebates are given to two classes of accounts for a specified period `P`:
- Voters who revealed their votes 
- Voters who claimed their rewards 

We break these into two separate classes because voters who revealed their votes incorrectly (i.e. not with the majority) should still get rebated.

Rebates are calculated using the actual gas limit spent by `reveal`, `commit`, and `claim` transactions and the gas price will be determined based on some average gas price over `P` and some average ETH/USD price over `P`. Rebates will be distributed in UMA.

Follow up PR will integrate the [Etherscan PRO API](https://etherscan.io/apis#gastracker) to calculate `P` and `ETH/USD` historical prices.

** Details **
Example run for votes from August 28 until September 2:

```sh
 node ./packages/core/scripts/gas-rebate/index.js  --start 1598572800 --end 1599055419
```

### CODE REVIEW

**Comment by nicholaspai** [comment]
Latest run with `cli-progress`:

```sh
➜  core git:(npai/gas-rebate-script) ✗ yarn truffle exec ./scripts/gas-rebate/index.js --network mainnet_mnemonic
yarn run v1.22.0
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/gas-rebate/index.js --network mainnet_mnemonic
Using network 'mainnet_mnemonic'.

⛽️ Calculating gas rebates from block 10606000 until 10745204
📸 Parsing REVEAL data:
[████████████████████████████████████████] 100% | reveal events parsed: 49/49
✅ Finished parsing REVEAL data.
💴 Parsing CLAIM data:
[████████████████████████████████████████] 100% | claim events parsed: 50/50
✅ Finished parsing CLAIM data.
49 Voters Revealed
REVEAL (0xBEe33aC5520a2245CBF8288e768A5cC26927cddF) gas: 437406 (UUID: 0xBEe33aC5520a2245CBF8288e768A5cC26927cddF-9240-USDETH-1596666977)
REVEAL (0x20F9E5a17b00350F1EAF9Ad18Bc5be0D54c0bc3F) gas: 174418 (UUID: 0x20F9E5a17b00350F1EAF9Ad18Bc5be0D54c0bc3F-9240-USDETH-1596666977)
REVEAL (0xb85dE2582A8A896c743DfFf9553Adc1B4473E416) gas: 163114 (UUID: 0xb85dE2582A8A896c743DfFf9553Adc1B4473E416-9240-USDETH-1596666977)
REVEAL (0x28206a7fd0c5b115754319f1eaD8BfEF2498de90) gas: 159166 (UUID: 0x28206a7fd0c5b115754319f1eaD8BfEF2498de90-9240-USDETH-1596666977)
REVEAL (0x188C30E9A6527F5F0c3f7fe59B72ac7253C62F28) gas: 139082 (UUID: 0x188C30E9A6527F5F0c3f7fe59B72ac7253C62F28-9240-USDETH-1596666977)
REVEAL (0x323AFc6d2DbE2Eb0AE71Ed1f87db8956834E861d) gas: 139082 (UUID: 0x323AFc6d2DbE2Eb0AE71Ed1f87db8956834E861d-9240-USDETH-1596666977)
REVEAL (0xF7B10D603907658F690Da534E9b7dbC4dAB3E2D6) gas: 143006 (UUID: 0xF7B10D603907658F690Da534E9b7dbC4dAB3E2D6-9240-USDETH-1596666977)
REVEAL (0x28d804Bf2212E220BC2B7B6252993Db8286dF07f) gas: 143006 (UUID: 0x28d804Bf2212E220BC2B7B6252993Db8286dF07f-9240-USDETH-1596666977)
REVEAL (0x8d680d96BEFC53D8B90F76d1e8E99e842b3387A2) gas: 163090 (UUID: 0x8d680d96BEFC53D8B90F76d1e8E99e842b3387A2-9240-USDETH-1596666977)
REVEAL (0x67D24f48f1B58C4d4DcDA57Ba2386DcE76353fA2) gas: 159190 (UUID: 0x67D24f48f1B58C4d4DcDA57Ba2386DcE76353fA2-9240-USDETH-1596666977)
REVEAL (0x61Ba66498732AE3b27a16F1193de77319D0DeEeb) gas: 159142 (UUID: 0x61Ba66498732AE3b27a16F1193de77319D0DeEeb-9240-USDETH-1596666977)
REVEAL (0x2117c37a65aD3c0489682386f7d81d4c6d08B3C8) gas: 139082 (UUID: 0x2117c37a65aD3c0489682386f7d81d4c6d08B3C8-9240-USDETH-1596666977)
REVEAL (0xBe3BF9Bb42bF423695B792250BE206C9A58e2dE1) gas: 143006 (UUID: 0xBe3BF9Bb42bF423695B792250BE206C9A58e2dE1-9240-USDETH-1596666977)
REVEAL (0x797E1f1765a0071454f443114Bf26a53F798c071) gas: 163114 (UUID: 0x797E1f1765a0071454f443114Bf26a53F798c071-9240-USDETH-1596666977)
REVEAL (0x2cE560f2CB8d58ABDC0126CDb24f6ee084058EF6) gas: 143006 (UUID: 0x2cE560f2CB8d58ABDC0126CDb24f6ee084058EF6-9240-USDETH-1596666977)
REVEAL (0x4c39185a078B5666C372538231cB793A0928807b) gas: 143006 (UUID: 0x4c39185a078B5666C372538231cB793A0928807b-9240-USDETH-1596666977)
REVEAL (0x2eA5b7c369cb88c171466Bb6dcf3Cbe1C508FBeA) gas: 167038 (UUID: 0x2eA5b7c369cb88c171466Bb6dcf3Cbe1C508FBeA-9240-USDETH-1596666977)
REVEAL (0x49a8a045EaE2C9D4a7C7e2919bA957064Bd59252) gas: 167038 (UUID: 0x49a8a045EaE2C9D4a7C7e2919bA957064Bd59252-9240-USDETH-1596666977)
REVEAL (0x40630B4Fd61523450b3dbc4d9Fd672AAE0A9c31C) gas: 139082 (UUID: 0x40630B4Fd61523450b3dbc4d9Fd672AAE0A9c31C-9240-USDETH-1596666977)
REVEAL (0x896b94f4f27f12369698C302e2049cAe86936BbB) gas: 143006 (UUID: 0x896b94f4f27f12369698C302e2049cAe86936BbB-9240-USDETH-1596666977)
REVEAL (0x49739691fB5f3992b3f2536f309D955558e75933) gas: 143006 (UUID: 0x49739691fB5f3992b3f2536f309D955558e75933-9240-USDETH-1596666977)
REVEAL (0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165) gas: 143006 (UUID: 0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165-9240-USDETH-1596666977)
REVEAL (0x9ae57ce31BDae73A0d3BF701998Af5494a1cEeec) gas: 139082 (UUID: 0x9ae57ce31BDae73A0d3BF701998Af5494a1cEeec-9240-USDETH-1596666977)
REVEAL (0xD2A78Bb82389D30075144d17E782964918999F7f) gas: 143006 (UUID: 0xD2A78Bb82389D30075144d17E782964918999F7f-9240-USDETH-1596666977)
REVEAL (0x120219511b9B0254E7D15014dB83B8752289FdA3) gas: 163114 (UUID: 0x120219511b9B0254E7D15014dB83B8752289FdA3-9240-USDETH-1596666977)
REVEAL (0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947) gas: 159190 (UUID: 0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947-9240-USDETH-1596666977)
REVEAL (0x6888FF4bE3262F3232e1c022De731680a16dDdA7) gas: 143006 (UUID: 0x6888FF4bE3262F3232e1c022De731680a16dDdA7-9240-USDETH-1596666977)
REVEAL (0xCcD9C5465faC5C0Df86fD9B7d524a5949dEFcef6) gas: 142982 (UUID: 0xCcD9C5465faC5C0Df86fD9B7d524a5949dEFcef6-9240-USDETH-1596666977)
REVEAL (0x0C30a911CfEe90172dba6533DA18593C9F04F807) gas: 139082 (UUID: 0x0C30a911CfEe90172dba6533DA18593C9F04F807-9240-USDETH-1596666977)
REVEAL (0xfEc2Ade141272f5271FEbaDFca5d21Ca30fc04a5) gas: 139082 (UUID: 0xfEc2Ade141272f5271FEbaDFca5d21Ca30fc04a5-9240-USDETH-1596666977)
REVEAL (0xEb6257954bb490556599E0f4E305051821a33152) gas: 139082 (UUID: 0xEb6257954bb490556599E0f4E305051821a33152-9240-USDETH-1596666977)
REVEAL (0xbac4ce8dD5E6F26166A8909912Db53a832948334) gas: 174418 (UUID: 0xbac4ce8dD5E6F26166A8909912Db53a832948334-9240-USDETH-1596666977)
REVEAL (0xF627E5F4BAd95a956468d8BB6Ee20b119F992E96) gas: 143006 (UUID: 0xF627E5F4BAd95a956468d8BB6Ee20b119F992E96-9240-USDETH-1596666977)
REVEAL (0x996d5138843993Cebc554c20d82310a43802aFF1) gas: 139082 (UUID: 0x996d5138843993Cebc554c20d82310a43802aFF1-9240-USDETH-1596666977)
REVEAL (0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A) gas: 142982 (UUID: 0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A-9240-USDETH-1596666977)
REVEAL (0xffd4DAe0d7D8ddb6F408dca0a47763Ae3A57F4Ce) gas: 139082 (UUID: 0xffd4DAe0d7D8ddb6F408dca0a47763Ae3A57F4Ce-9240-USDETH-1596666977)
REVEAL (0xf86eb5215F14AF0eC2708653dC8A7c6714648Eb1) gas: 174442 (UUID: 0xf86eb5215F14AF0eC2708653dC8A7c6714648Eb1-9240-USDETH-1596666977)
REVEAL (0xD7925984736824B4Aecf8301c9aEE211Cd976494) gas: 143006 (UUID: 0xD7925984736824B4Aecf8301c9aEE211Cd976494-9240-USDETH-1596666977)
REVEAL (0x8072D014d5F8d5904A1B71E6390d93C4D03D58f3) gas: 139082 (UUID: 0x8072D014d5F8d5904A1B71E6390d93C4D03D58f3-9240-USDETH-1596666977)
REVEAL (0x522c2AeB8488b0b5ca22B803d0A2945f5B0482B2) gas: 163114 (UUID: 0x522c2AeB8488b0b5ca22B803d0A2945f5B0482B2-9240-USDETH-1596666977)
REVEAL (0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48) gas: 143006 (UUID: 0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48-9240-USDETH-1596666977)
REVEAL (0x14A29ccfE3C1cc777A6f3a2ead6F07ceC5cC5dc1) gas: 167038 (UUID: 0x14A29ccfE3C1cc777A6f3a2ead6F07ceC5cC5dc1-9240-USDETH-1596666977)
REVEAL (0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615) gas: 167038 (UUID: 0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615-9240-USDETH-1596666977)
REVEAL (0xf0A04b2E7965E778e298Fe0D234d132fe28FCfd1) gas: 139082 (UUID: 0xf0A04b2E7965E778e298Fe0D234d132fe28FCfd1-9240-USDETH-1596666977)
REVEAL (0x966Cf5cd0624f1EfCf21B0abc231A5CcC802B861) gas: 139058 (UUID: 0x966Cf5cd0624f1EfCf21B0abc231A5CcC802B861-9240-USDETH-1596666977)
REVEAL (0x8729e0f81Bfa27d0fa250Bc05517E161090AB051) gas: 142982 (UUID: 0x8729e0f81Bfa27d0fa250Bc05517E161090AB051-9240-USDETH-1596666977)
REVEAL (0x10aab4B0EF76AA2AC9b5909e671517a1171B050E) gas: 139082 (UUID: 0x10aab4B0EF76AA2AC9b5909e671517a1171B050E-9240-USDETH-1596666977)
REVEAL (0xe311958aC79b41BF3a0595470A964B8EFBe0fb36) gas: 167038 (UUID: 0xe311958aC79b41BF3a0595470A964B8EFBe0fb36-9240-USDETH-1596666977)
REVEAL (0x1B52A70616Ee7ed4Ea070d19FD18a60e467a5276) gas: 143006 (UUID: 0x1B52A70616Ee7ed4Ea070d19FD18a60e467a5276-9240-USDETH-1596666977)
50 Voters Claimed Rewards
CLAIM (0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48) gas: 92086 (UUID: 0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48-9238-ETH/BTC-1596240000)
CLAIM (0x28d804Bf2212E220BC2B7B6252993Db8286dF07f) gas: 93690 (UUID: 0x28d804Bf2212E220BC2B7B6252993Db8286dF07f-9238-COMPUSD-1596240000)
CLAIM (0x28d804Bf2212E220BC2B7B6252993Db8286dF07f) gas: 93690 (UUID: 0x28d804Bf2212E220BC2B7B6252993Db8286dF07f-9238-ETH/BTC-1596240000)
CLAIM (0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48) gas: 195834 (UUID: 0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48-9238-COMPUSD-1596240000)
CLAIM (0xEb6257954bb490556599E0f4E305051821a33152) gas: 178945 (UUID: 0xEb6257954bb490556599E0f4E305051821a33152-9240-USDETH-1596666977)
CLAIM (0xfEc2Ade141272f5271FEbaDFca5d21Ca30fc04a5) gas: 141245 (UUID: 0xfEc2Ade141272f5271FEbaDFca5d21Ca30fc04a5-9240-USDETH-1596666977)
CLAIM (0x996d5138843993Cebc554c20d82310a43802aFF1) gas: 141245 (UUID: 0x996d5138843993Cebc554c20d82310a43802aFF1-9240-USDETH-1596666977)
CLAIM (0x0C30a911CfEe90172dba6533DA18593C9F04F807) gas: 141245 (UUID: 0x0C30a911CfEe90172dba6533DA18593C9F04F807-9240-USDETH-1596666977)
CLAIM (0xF7B10D603907658F690Da534E9b7dbC4dAB3E2D6) gas: 143207 (UUID: 0xF7B10D603907658F690Da534E9b7dbC4dAB3E2D6-9240-USDETH-1596666977)
CLAIM (0x8d680d96BEFC53D8B90F76d1e8E99e842b3387A2) gas: 143207 (UUID: 0x8d680d96BEFC53D8B90F76d1e8E99e842b3387A2-9240-USDETH-1596666977)
CLAIM (0x797E1f1765a0071454f443114Bf26a53F798c071) gas: 92074 (UUID: 0x797E1f1765a0071454f443114Bf26a53F798c071-9240-USDETH-1596666977)
CLAIM (0x40630B4Fd61523450b3dbc4d9Fd672AAE0A9c31C) gas: 141245 (UUID: 0x40630B4Fd61523450b3dbc4d9Fd672AAE0A9c31C-9240-USDETH-1596666977)
CLAIM (0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947) gas: 142865 (UUID: 0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947-9238-COMPUSD-1596240000)
CLAIM (0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947) gas: 142865 (UUID: 0xb65A3bd2e3f4Cc4AC58FcBafe0E6e06359994947-9238-ETH/BTC-1596240000)
CLAIM (0x49739691fB5f3992b3f2536f309D955558e75933) gas: 143207 (UUID: 0x49739691fB5f3992b3f2536f309D955558e75933-9240-USDETH-1596666977)
CLAIM (0x120219511b9B0254E7D15014dB83B8752289FdA3) gas: 143207 (UUID: 0x120219511b9B0254E7D15014dB83B8752289FdA3-9240-USDETH-1596666977)
CLAIM (0xF627E5F4BAd95a956468d8BB6Ee20b119F992E96) gas: 143207 (UUID: 0xF627E5F4BAd95a956468d8BB6Ee20b119F992E96-9240-USDETH-1596666977)
CLAIM (0x188C30E9A6527F5F0c3f7fe59B72ac7253C62F28) gas: 141245 (UUID: 0x188C30E9A6527F5F0c3f7fe59B72ac7253C62F28-9240-USDETH-1596666977)
CLAIM (0x2117c37a65aD3c0489682386f7d81d4c6d08B3C8) gas: 141245 (UUID: 0x2117c37a65aD3c0489682386f7d81d4c6d08B3C8-9240-USDETH-1596666977)
CLAIM (0x1B52A70616Ee7ed4Ea070d19FD18a60e467a5276) gas: 143207 (UUID: 0x1B52A70616Ee7ed4Ea070d19FD18a60e467a5276-9240-USDETH-1596666977)
CLAIM (0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A) gas: 55263 (UUID: 0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A-9231-Admin 8-1595095182)
CLAIM (0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A) gas: 55263 (UUID: 0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A-9231-Admin 9-1595095665)
CLAIM (0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A) gas: 143207 (UUID: 0xDb2808ed6cF0B8B9686222689CF4f6e88E42B80A-9240-USDETH-1596666977)
CLAIM (0x8729e0f81Bfa27d0fa250Bc05517E161090AB051) gas: 143207 (UUID: 0x8729e0f81Bfa27d0fa250Bc05517E161090AB051-9240-USDETH-1596666977)
CLAIM (0x9ae57ce31BDae73A0d3BF701998Af5494a1cEeec) gas: 141245 (UUID: 0x9ae57ce31BDae73A0d3BF701998Af5494a1cEeec-9240-USDETH-1596666977)
CLAIM (0xD2A78Bb82389D30075144d17E782964918999F7f) gas: 143195 (UUID: 0xD2A78Bb82389D30075144d17E782964918999F7f-9240-USDETH-1596666977)
CLAIM (0xCcD9C5465faC5C0Df86fD9B7d524a5949dEFcef6) gas: 143207 (UUID: 0xCcD9C5465faC5C0Df86fD9B7d524a5949dEFcef6-9240-USDETH-1596666977)
CLAIM (0x28d804Bf2212E220BC2B7B6252993Db8286dF07f) gas: 143207 (UUID: 0x28d804Bf2212E220BC2B7B6252993Db8286dF07f-9240-USDETH-1596666977)
CLAIM (0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165) gas: 58593 (UUID: 0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165-9231-Admin 8-1595095182)
CLAIM (0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165) gas: 143207 (UUID: 0xff4e2D7169D7D56f8b41C63592FA919b4d1B0165-9240-USDETH-1596666977)
CLAIM (0xD7925984736824B4Aecf8301c9aEE211Cd976494) gas: 143207 (UUID: 0xD7925984736824B4Aecf8301c9aEE211Cd976494-9240-USDETH-1596666977)
CLAIM (0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48) gas: 92074 (UUID: 0xEafdcDc6B92e93Cc15dDbD452Cf20f760cd2Ce48-9240-USDETH-1596666977)
CLAIM (0x896b94f4f27f12369698C302e2049cAe86936BbB) gas: 143207 (UUID: 0x896b94f4f27f12369698C302e2049cAe86936BbB-9240-USDETH-1596666977)
CLAIM (0xBEe33aC5520a2245CBF8288e768A5cC26927cddF) gas: 143207 (UUID: 0xBEe33aC5520a2245CBF8288e768A5cC26927cddF-9240-USDETH-1596666977)
CLAIM (0x522c2AeB8488b0b5ca22B803d0A2945f5B0482B2) gas: 143207 (UUID: 0x522c2AeB8488b0b5ca22B803d0A2945f5B0482B2-9240-USDETH-1596666977)
CLAIM (0xffd4DAe0d7D8ddb6F408dca0a47763Ae3A57F4Ce) gas: 141245 (UUID: 0xffd4DAe0d7D8ddb6F408dca0a47763Ae3A57F4Ce-9240-USDETH-1596666977)
CLAIM (0x67D24f48f1B58C4d4DcDA57Ba2386DcE76353fA2) gas: 141245 (UUID: 0x67D24f48f1B58C4d4DcDA57Ba2386DcE76353fA2-9240-USDETH-1596666977)
CLAIM (0x10aab4B0EF76AA2AC9b5909e671517a1171B050E) gas: 141245 (UUID: 0x10aab4B0EF76AA2AC9b5909e671517a1171B050E-9240-USDETH-1596666977)
CLAIM (0x49a8a045EaE2C9D4a7C7e2919bA957064Bd59252) gas: 145169 (UUID: 0x49a8a045EaE2C9D4a7C7e2919bA957064Bd59252-9240-USDETH-1596666977)
CLAIM (0x2eA5b7c369cb88c171466Bb6dcf3Cbe1C508FBeA) gas: 145169 (UUID: 0x2eA5b7c369cb88c171466Bb6dcf3Cbe1C508FBeA-9240-USDETH-1596666977)
CLAIM (0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615) gas: 58593 (UUID: 0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615-9231-Admin 5-1595094121)
CLAIM (0x14A29ccfE3C1cc777A6f3a2ead6F07ceC5cC5dc1) gas: 145169 (UUID: 0x14A29ccfE3C1cc777A6f3a2ead6F07ceC5cC5dc1-9240-USDETH-1596666977)
CLAIM (0x4c39185a078B5666C372538231cB793A0928807b) gas: 143207 (UUID: 0x4c39185a078B5666C372538231cB793A0928807b-9240-USDETH-1596666977)
CLAIM (0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615) gas: 145169 (UUID: 0xBf87C98F8A9c1F2A88b613813Ec1E974105EB615-9240-USDETH-1596666977)
CLAIM (0x6888FF4bE3262F3232e1c022De731680a16dDdA7) gas: 143207 (UUID: 0x6888FF4bE3262F3232e1c022De731680a16dDdA7-9240-USDETH-1596666977)
CLAIM (0xBe3BF9Bb42bF423695B792250BE206C9A58e2dE1) gas: 92074 (UUID: 0xBe3BF9Bb42bF423695B792250BE206C9A58e2dE1-9240-USDETH-1596666977)
CLAIM (0x61Ba66498732AE3b27a16F1193de77319D0DeEeb) gas: 141245 (UUID: 0x61Ba66498732AE3b27a16F1193de77319D0DeEeb-9240-USDETH-1596666977)
CLAIM (0x28206a7fd0c5b115754319f1eaD8BfEF2498de90) gas: 142865 (UUID: 0x28206a7fd0c5b115754319f1eaD8BfEF2498de90-9238-COMPUSD-1596240000)
CLAIM (0x28206a7fd0c5b115754319f1eaD8BfEF2498de90) gas: 142865 (UUID: 0x28206a7fd0c5b115754319f1eaD8BfEF2498de90-9238-ETH/BTC-1596240000)
CLAIM (0x2cE560f2CB8d58ABDC0126CDb24f6ee084058EF6) gas: 143207 (UUID: 0x2cE560f2CB8d58ABDC0126CDb24f6ee084058EF6-9240-USDETH-1596666977)
✨  Done in 29.14s.
```

File: packages/core/scripts/liquidity-mining/FindBlockAtTimeStamp.js [hunk 1]
```diff
@@ -98,3 +96,6 @@ async function _findBlockNumberAtTimestamp(web3, targetTimestamp, higherLimitMax
   }
   return { blockNumber: block.number, error: Math.abs(targetTimestamp - block.timestamp) };
 }
+
+FindBlockAtTimeStamp._findBlockNumberAtTimestamp = _findBlockNumberAtTimestamp;
```
Comment (nicholaspai) [inline packages/core/scripts/liquidity-mining/FindBlockAtTimeStamp.js:100]: @chrismaree just want to highlight this change. I want to access the very useful helper function `_findBlockNumberAtTImestamp` in the gas rebate script, so I added it to the exports.

File: packages/core/scripts/gas-rebate/index.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
+const cliProgress = require("cli-progress");
```
Comment (chrismaree) [inline packages/core/scripts/gas-rebate/index.js:1]: Can you please add a comment explaining what this script does and how to run it up at the top?

File: packages/core/scripts/gas-rebate/index.js [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["start", "end"],
+  boolean: ["reveal-only", "claim-only"]
+});
+const fs = require("fs");
+const path = require("path");
+
+const Voting = artifacts.require("Voting");
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply packages/core/scripts/gas-rebate/index.js:9]: you'll also need to change how you get your web3 instance for now you can import the web3 instance like [this](https://github.com/UMAprotocol/protocol/blob/319610ed7d6e92e289e37014ee6a8212b2d977ad/packages/core/scripts/liquidity-mining/CalculateBalancerLPRewards.js#L15) and when PR #1928 and #1917 are merged in you can use the `getWeb3` from `require("@umaprotocol/common");`

File: packages/core/scripts/gas-rebate/index.js [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      voting.getPastEvents("RewardsRetrieved", {
+        fromBlock: startBlock,
+        toBlock: endBlock
+      })
+    ]);
+
+    // TODO: Fetch gas price data
+    const SCALING_FACTOR = toBN(toWei("1"));
+    // - Get gas price for period.  This is the ETH price per unit gas, described in Gwei.
+    const _averagePriceGweiForPeriod = "90";
```
Comment (chrismaree) [inline packages/core/scripts/gas-rebate/index.js:49]: if only the average was 90. I guess this is why the total gas rebate was only ~500 usd.

File: packages/core/scripts/gas-rebate/index.js [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+    console.log("\n\n*=======================================*");
+    console.log("*                                       *");
+    console.log("* 🧮 Final UMA Rebate                   *");
+    console.log("*                                       *");
+    console.log("*=======================================*");
+    console.log(`🎟 UMA to rebate: ${totalUMAToRebate}`);
+    console.log(`📒 Output JSON: ${JSON.stringify(rebateOutput, null, 4)}`);
+
+    // Format output and save to file.
+    const savePath = `${path.resolve(__dirname)}/weekly-payouts/Week_${weekNumber}_Gas_Rebate.json`;
```
Comment (chrismaree) [inline packages/core/scripts/gas-rebate/index.js:291]: nit: is this a `weekly-payout`? I think a better name could be based off the voting rounds prehaps?

Comment (chrismaree) [reply packages/core/scripts/gas-rebate/index.js:291]: also you might consider adding additional information to the output such as the voting round, total number of commits, total number of revelas ect. if you'd like.

**Review by chrismaree — COMMENTED** [review]
This is pretty awesome! Nice and clean code & well commented. I have a few questions/comments and nits, mainly around how the script is run. Ideally this should be runnable as a node process, just like the Liquidity minding script.

File: packages/core/scripts/gas-rebate/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:9]: sure i'm good with this. I'll make the change to migrate to `node`

File: packages/core/scripts/gas-rebate/index.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+});
+const fs = require("fs");
+const path = require("path");
+
+const Voting = artifacts.require("Voting");
+
+const { toBN, toWei, fromWei } = web3.utils;
+const FindBlockAtTimestamp = require("../liquidity-mining/FindBlockAtTimeStamp");
+
+const CalculateRebate = async callback => {
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:13]: Good idea, will change

File: packages/core/scripts/gas-rebate/index.js (hunk 2 above)
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:2]: Yup

File: packages/core/scripts/gas-rebate/index.js (hunk 4 above)
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:74]: yep, these are just example hardcoded values for now and the final script will use the Etherscan API .

File: packages/core/scripts/gas-rebate/index.js (hunk 5 above)
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:316]: True, I'll just replace Week with Round

**Comment by nicholaspai** [comment]
I now parse Reveal and Claim events in parallel which should cut time in half by 50% on larger period queries

File: packages/core/scripts/gas-rebate/index.js [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+const web3 = new Web3(new Web3.providers.HttpProvider(process.env.CUSTOM_NODE_URL));
+const { toBN, toWei, fromWei } = web3.utils;
+
+async function parseRevealEvents(committedVotes, revealedVotes, priceData, multibar, rebateOutput) {
+  const revealVotersToRebate = {};
+
+  const progressBarReveal = multibar.create(revealedVotes.length, 0, { label: "Reveal Events" });
+
+  for (let i = 0; i < revealedVotes.length; i++) {
+    const reveal = revealedVotes[i];
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply packages/core/scripts/gas-rebate/index.js:46]: Good idea I'll refactor

**Pull request merged**